props/saber_BC_buttons.h

```diff
@@ -477,738 +1495,1272 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         }
       }
     }
-    // EVENT_THRUST
-      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
-          mss.x > 14  &&
-          fusor.swing_speed() < 150) {
-        if (millis() - thrust_begin_millis_ > 15) {
-          Event(BUTTON_NONE, EVENT_THRUST);
-          thrust_begin_millis_ = millis();
-        }
-      } else {
+// EVENT_THRUST
+    if (mss.y * mss.y + mss.z * mss.z < 20.0 &&  // slightly more forgiving than 16.0?
+       (mss.x > 14 || mss.x < -14) &&  // Check for both positive and negative x-motion
+        fusor.swing_speed() < 150) {
+      if (millis() - thrust_begin_millis_ > 15) {
+        Event(BUTTON_NONE, EVENT_THRUST);
         thrust_begin_millis_ = millis();
       }
-
-    // Mute timer check - play optional mute.wav first.
-    if (mute_timer_.timerCheck()) {
-      DoMute();
+    } else {
+      thrust_begin_millis_ = millis();
+    }
+// Timers
+    // Play optional mute.wav first.
+    if (mute_all_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnOnHelper();
+      }
+    }
+#ifdef BC_DUAL_BLADES
+    if (mute_mainBlade_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        PVLOG_NORMAL << "** Main Blade Turned ON Muted\n";
+      }
+    }
+    if (mute_secondBlade_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        muted_ = true;
+        TurnBladeOn(BC_SECOND_BLADE_SET);
+        PVLOG_NORMAL << "** Second Blade Turned ON Muted\n";
+      }
     }
-    // Scroll Presets timer check - avoid beep/wav overlap
+#endif
+    // Allow beep to be heard.
     if (scroll_presets_timer_.timerCheck() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
+    if (twist_timer_.timerCheck() && saved_twist_) {
```

Comment:
I'm not sure why you have to call timerCheck AND saved_twist_ here.
How can timerCheck return true if saved_twist_ is false?

Reply:
ah true. Timer wouldn't be running if no saved_twist_ is set, so there's no need to check both .
Done.